zkml/src/layers/transformer/qkv.rs

```diff
@@ -19,13 +38,49 @@ use crate::{
 /// the full K and V matrices as if they were computed using the whole input tensor.
 #[derive(Debug, Clone)]
 pub struct QKV<N> {
-    pub q: Tensor<N>,
-    pub q_bias: Tensor<N>,
-    pub k: Tensor<N>,
-    pub k_bias: Tensor<N>,
-    pub v: Tensor<N>,
-    pub v_bias: Tensor<N>,
-    // pub cache: Option<CacheQKV<N>>,
+    pub(crate) q: Tensor<N>,
+    pub(crate) q_bias: Tensor<N>,
+    pub(crate) k: Tensor<N>,
+    pub(crate) k_bias: Tensor<N>,
+    pub(crate) v: Tensor<N>,
+    pub(crate) v_bias: Tensor<N>,
+    unpadded_shape: Vec<usize>, /* same shape for Q, K and V
+                                 * pub cache: Option<CacheQKV<N>>, */
+}
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct QKVCtx<E> {
+    node_id: NodeId,
+    sumcheck_poly_aux: VPAuxInfo<E>,
+    padded_matrix_shape: Vec<usize>, // same shape for Q, K and V
+    unpadded_shape: Vec<usize>,      // same shape for Q, K and V
```

Comment:
Can't `padded_matrix_shape` be computed from `unpadded_shape` ? 
The distinction between the two was mostly useful for Conv where padded shape was way more complicated than simply `shape.iter().map(|d| d.next_power_of_two()).collect()` but i believe we _can_ use this simplification here, no ? Which should be quite streamlined/easy after we start using more and more the `Shape` struct i introduced in this [PR](https://github.com/Lagrange-Labs/deep-prove/pull/141).

Reply:
Done in commit https://github.com/Lagrange-Labs/deep-prove/pull/142/commits/bbd0c571172e9609a1a806186c96ab284b361f21